freeqdsk/geqdsk.py

```diff
@@ -182,6 +183,163 @@
+    zdim:
+        Height of computational domain in Z direction, float [meter]
+    rcentr:
+        Reference value of R, float [meter]
+    rleft:
+        R at left (inner) boundary, float [meter]
+    zmid:
+        Z at middle of domain, float [meter]
+    rmagx, rmaxis:
+        R at magnetic axis (0-point), float [meter]
+    zmagx, zmaxis:
+        Z at magnetic axis (0-point), float [meter]
+    simagx, psi_axis:
+        Poloidal flux :math:`\psi` at magnetic axis, float [weber / radian]
+    sibdry, psi_boundary:
+        Poloidal flux :math:`\psi` at plasma boundary, float [weber / radian]
+    bcentr:
+        Vacuum toroidal magnetic field at rcentr, float [tesla]
+    cpasma, current:
+        Plasma current, float [ampere]
+    fpol:
+        Poloidal current function :math:`F(\psi)=RB_t`, 1D array [meter * tesla]
+    pres, pressure:
+        Plasma pressure :math:`p(\psi)`, 1D array [pascal]
+    ffprime:
+        :math:`FF'(\psi)`, 1D array [meter**2 * tesla**2 * radian / weber]
+    pprime:
+        :math:`p'(\psi)`, 1D array [pascal * radian / weber]
+    psi, f, psirz:
+         Poloidal flux :math:`\psi`, 2D array [weber / radian]
+    qpsi:
+        Safety factor :math:`q(\psi)`, 1D array [dimensionless]
+    nbdry, nbbbs:
+        Number of points in the boundary grid, int
+    nlim, limitr:
+        Number of points in the limiter grid, int
+    rbdry, rbbbs:
+        R of boundary points, 1D array [meter]
+    zbdry, zbbbs:
+        Z of boundary points, 1D array [meter]
+    rlim:
+        R of limiter points, 1D array [meter]
+    zlim:
+        Z of limiter points, 1D array [meter]
+
+    Examples
+    --------
+
+    Variables can be accessed with either the attribute dot syntax or ``dict``
+    item access. For example, the poloidal flux on the magnetic axis can be
+    accessed via any one of the following:
+
+    >>> gfile = GEQDSKFile(...)
+    >>> print(gfile["simagx"])
+    >>> print(gfile.simagx)
+    >>> print(gfile.psi_axis)
+
+    """
+
+    comment: str
+    shot: int
+    nx: int
+    ny: int
+    rdim: float
+    zdim: float
+    rcentr: float
+    rleft: float
+    zmid: float
+    rmagx: float
+    zmagx: float
+    simagx: float
+    sibdry: float
+    bcentr: float
+    cpasma: float
+    fpol: FloatArray
+    pres: FloatArray
+    ffprime: FloatArray
+    pprime: FloatArray
+    psi: FloatArray
+    qpsi: FloatArray
+    nbdry: int
+    nlim: int
+    rbdry: FloatArray | None = None
+    zbdry: FloatArray | None = None
+    rlim: FloatArray | None = None
+    zlim: FloatArray | None = None
+
+    def __post_init__(self):
+        r = np.zeros((self.nx, self.ny), float)
+        z = np.zeros((self.nx, self.ny), float)
+
+        for i in range(0, self.nx):
+            for j in range(0, self.ny):
+                r[i, j] = self.rleft + self.rdim * i / (self.nx - 1)
+                z[i, j] = (self.zmid - 0.5 * self.zdim) + self.zdim * j / (self.ny - 1)
```

Comment:
I know performance isn't the top priority here, but the following should do it without the for loops:

```python
r = self.rleft + self.rdim * np.arange(self.nx) / (self.nx - 1)
z = (self.zmin - 0.5 * self.zdim) + self.zdim * np.arange(self.ny) / (self.ny - 1)
self.r_grid, self.z_grid = np.meshgrid(r, z, indexing="ij")
```
```

Reply:
Thanks! This was just copy-pasted from another geqdsk reader